services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/DirectionsJsonObject.java

```diff
@@ -1,18 +1,22 @@
 package com.mapbox.api.directions.v5.models;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 import com.google.gson.GsonBuilder;
 import com.mapbox.api.directions.v5.DirectionsAdapterFactory;
 import com.mapbox.geojson.Point;
 import com.mapbox.geojson.PointAsCoordinatesTypeAdapter;
+import com.ryanharter.auto.value.gson.UnrecognisedJsonProperties;
 
 import java.io.Serializable;
+import java.util.Map;
 
 /**
  * Provides a base class for Directions model classes.
  *
  * @since 3.4.0
  */
-public class DirectionsJsonObject implements Serializable {
+public abstract class DirectionsJsonObject implements Serializable {
```

Comment:
This is a breaking change (not sure how impactful it would actually be in practice). Could we make the methods return empty and implement downstream, how would it look like?

Reply:
nice catch 👍

Reply:
AutoValue won't generate fields for not abstract methods

Reply:
I see a few possible solutions right now. 

### Do breaking change

Maybe we can make ourself believe that it's not really a breaking change using following arguments: it more like a fix of a bug, nobody should have created an instance of `DirectionsJsonObject`, it doesn't make any sense to create an instance of `DirectionsJsonObject`. But `abstract` do really changes API.

### Workaround

I can create an abstract class `BaseDirectionsJsonObject` which is going to be a parent of `DirectionsJsonObject` and move `unrecognised` filed there. I find this solution ugly but it won't break API.

Reply:
I think that we need to decide if somebody could be affected by not being able to instantiate `DirectionsJsonObject`. If we're sure that we will affect nobody, and even if we will we're ready to reply in their issues on github, then I let's break the rule.
If we don't feel confident, let's workaround

Reply:
I feel confident. I can't imaging a case when somebody instantiate `DirectionsJsonObject` but if somebody does it, I'd like to learn their case when they open an issue on our github 🙂 
@LukasPaczos , @RingerJK , what do you think about it?

Reply:
@VysotskiVadim I agree. There's no practical reason to extend the `DirectionsJsonObject` for developers, I think we can break here. Let's clearly state this in the changelog and go from there.